File: https-quera.org-problemset-49028/Solution.py
def app():
    nums = []
    res = 0
    for _ in range(int(input())):
        nums.append(int(input()))
    last = nums[0]
    for _ in nums:
        if _ != last:
            last = _
            res += 1

    print(res)


if __name__ == "__main__":
    app()

File: https-quera.org-problemset-49028/Solution.java
import java.util.ArrayList;
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        app();
    }

    public static void app() {
        Scanner scanner = new Scanner(System.in);
        ArrayList<Integer> nums = new ArrayList<>();
        int res = 0;

        int n = scanner.nextInt();
        for (int i = 0; i < n; i++)
            nums.add(scanner.nextInt());

        if (!nums.isEmpty()) {
            int last = nums.get(0);

            for (int num : nums) {
                if (num != last) {
                    last = num;
                    res++;
                }
            }
        }

        System.out.println(res);
    }
}
